thetis/utility.py

```diff
@@ -884,18 +903,11 @@ def __init__(self, input_2d, output_3d, elem_height=None):
         self.fs_3d = self.output_3d.function_space()
 
         family_2d = self.fs_2d.ufl_element().family()
-        ufl_elem = self.fs_3d.ufl_element()
-        if isinstance(ufl_elem, ufl.VectorElement):
-            # Unwind vector
-            ufl_elem = ufl_elem.sub_elements()[0]
-        if isinstance(ufl_elem, ufl.HDivElement):
-            # RT case
-            ufl_elem = ufl_elem._element
-        if ufl_elem.family() == 'TensorProductElement':
-            # a normal tensorproduct element
-            family_3dh = ufl_elem.sub_elements()[0].family()
-            if family_2d != family_3dh:
-                raise Exception('2D and 3D spaces do not match: {0:s} {1:s}'.format(family_2d, family_3dh))
+        base_element_3d = get_extruded_base_element(self.fs_3d.ufl_element())
+        assert isinstance(base_element_3d, ufl.TensorProductElement)
+        family_3dh = base_element_3d.sub_elements()[0].family()
+        if family_2d != family_3dh:
+            raise Exception('2D and 3D spaces do not match: {0:s} {1:s}'.format(family_2d, family_3dh))
```

Comment:
Could maybe write another utility function called something like `check_element_consistency_2d-3d`, since this is repeated a couple of times later?

Reply:
Yeah, sure. Will do in another PR.